### PR TITLE
ACS-8729: Adding ACS health check probe.

### PR DESCRIPTION
**Pull request by mpichura**

### CODE REVIEW

File: prediction-applier/src/main/java/org/alfresco/hxi_connector/prediction_applier/config/RepositoryApiProperties.java [hunk 1]
```diff
@@ -25,10 +25,29 @@
...
-public record RepositoryApiProperties(String baseUrl, String discoveryEndpoint, Retry retry)
-{}
+public record RepositoryApiProperties(String baseUrl, String discoveryEndpoint, Retry retry, HealthProbe healthProbe)
+{
+
+    public RepositoryApiProperties
+    {
+        if (StringUtils.isNotBlank(discoveryEndpoint) && StringUtils.isBlank(healthProbe.endpoint()))
+        {
+            throw new IllegalStateException("Property %s must be set in the Live Ingester configuration when property %s is set."
```
Comment (krdabrowski) [inline prediction-applier/src/main/java/org/alfresco/hxi_connector/prediction_applier/config/RepositoryApiProperties.java:45]:
```suggestion
            throw new IllegalStateException("Property %s must be set in the Prediction Applier configuration when property %s is set."
```

**Review by krdabrowski — APPROVED** [review]
LGTM

File: prediction-applier/src/main/java/org/alfresco/hxi_connector/prediction_applier/config/RepositoryApiProperties.java (hunk 1 above)
Comment (mpichura) [reply prediction-applier/src/main/java/org/alfresco/hxi_connector/prediction_applier/config/RepositoryApiProperties.java]: Thanks, fixed

**Review by Pawel-608 — APPROVED** [review]
LGTM

**Pull request merged**